File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n= nums.length;

        HashMap<Integer, Integer> hm= new HashMap<>();
        for(int i=0;i<n ;i++){
            int diff= target-nums[i];

            if(hm.containsKey(diff)){
                return new int[]{i, hm.get(diff)};
            }
            hm.put(nums[i],i);
        }
        return new int[]{-1,-1};
    }
}

File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map={}
        sum=0
        for ind, num in enumerate(nums):
            sum= target-num
            if sum  in map:
                return [ind, map[sum]]
            map[num]=ind
        return [-1,-1]

        